app/src/main/java/com/epfl/beatlink/viewmodel/library/PlaylistViewModel.kt

```diff
@@ -204,10 +204,22 @@ class PlaylistViewModel(
     ?: emptyList() // Return an empty list if no playlist is selected
   }
 
-  fun deleteOwnedPlaylists() {
-    repository.deleteOwnedPlaylists(
-        onSuccess = { Log.d("PlaylistViewModel", "All playlists deleted successfully") },
-        onFailure = { e -> Log.e("PlaylistViewModel", "Failed to delete playlists", e) })
+  fun deleteOwnedPlaylists(): Boolean {
```

Comment:
Needs to treat the case where the data does not exist. Needs to not give a failure on non existence of the data.